src/globalInfo/dxEnvVars.ts

```diff
@@ -0,0 +1,371 @@
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_INSTANCE_URL)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_JSON_TO_STDOUT]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_JSON_TO_STDOUT)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_LOG_LEVEL]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_LOG_LEVEL)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_MAX_QUERY_LIMIT]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_MAX_QUERY_LIMIT)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_MDAPI_TEMP_DIR]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_MDAPI_TEMP_DIR)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_NPM_REGISTRY]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_NPM_REGISTRY)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_PRECOMPILE_ENABLE]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_PRECOMPILE_ENABLE)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_PROJECT_AUTOUPDATE_DISABLE_FOR_PACKAGE_CREATE]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_PROJECT_AUTOUPDATE_DISABLE_FOR_PACKAGE_CREATE)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_PROJECT_AUTOUPDATE_DISABLE_FOR_PACKAGE_VERSION_CREATE]: {
+    description: messages.getMessage(
+      camelCase(DxWellKnownEnvVars.SF_PROJECT_AUTOUPDATE_DISABLE_FOR_PACKAGE_VERSION_CREATE)
+    ),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_REST_DEPLOY]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_REST_DEPLOY)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_SOURCE_MEMBER_POLLING_TIMEOUT]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_SOURCE_MEMBER_POLLING_TIMEOUT)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_USE_GENERIC_UNIX_KEYCHAIN]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_USE_GENERIC_UNIX_KEYCHAIN)),
+    synonymOf: undefined,
+  },
+  [DxWellKnownEnvVars.SF_USE_PROGRESS_BAR]: {
+    description: messages.getMessage(camelCase(DxWellKnownEnvVars.SF_USE_PROGRESS_BAR)),
+    synonymOf: undefined,
+  },
+};
+
+export const envVarsResolve = (): Dictionary<string> => {
+  const dict = {} as Dictionary<string>;
+  Object.entries(process.env).forEach(([key, value]) => {
+    // save all env vars to dictionary
+    dict[key] = value;
+    // cross populate value to synonym if synonym is null or undefined
+    if (DX_SUPPORTED_ENV_VARS[key]) {
+      if (DX_SUPPORTED_ENV_VARS[key].synonymOf) {
+        const synonym = DX_SUPPORTED_ENV_VARS[key].synonymOf;
+        // set synonym only if it is not set in in current env
+        if (synonym && Global.SFDX_INTEROPERABILITY) {
+          dict[synonym] = process.env[key];
+          process.env[synonym] = dict[synonym];
```

Comment:
In addition to placing the synonym value into the dictionary, also set process.env